File: src/main/java/com/sao/exception/GlobalExceptionHandler.java
package com.sao.exception;


import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.servlet.ModelAndView;

import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;

@ControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(value = Exception.class)
    public ModelAndView defaultErrorHandler(HttpServletRequest req, Exception e) throws Exception {
        ModelAndView mav = new ModelAndView();
        int code = 404;
        String message = "";
        if(e instanceof CException){
            message = ((CException)e).message;
            code = ((CException)e).code;
        }
        mav.addObject("errorCode", code);
        mav.addObject("errorMessage",message);
//        mav.addObject("url", req.getRequestURL());
        mav.setViewName("error");
        return mav;
    }

}

File: src/main/java/com/sao/exception/GlobalExceptionHandler.kt
package com.sao.exception

import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.servlet.ModelAndView
import java.lang.Exception
import javax.servlet.http.HttpServletRequest


@ControllerAdvice
class GlobalExceptionHandler{

    @ExceptionHandler(value = Exception::class)
    fun defaultErrorHandler( request: HttpServletRequest,e:Exception)
    :ModelAndView{
        var modelView = ModelAndView()
        modelView.addObject("exception",e)
        modelView.addObject("url",request.requestURL)
        modelView.viewName = "error"
        return modelView
    }
}